File: fixtures/scale/php/large/src/Families/LcgStream/LcgStreamDescriptor.php
<?php

namespace CipherLab\Families\LcgStream;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together LcgStreamCipher and
// LcgStreamVectors under the "stream" suite.
class LcgStreamDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'lcgstream';
    }

    public function suite(): string
    {
        return 'stream';
    }

    public function cipher(): Cipher
    {
        return new LcgStreamCipher();
    }

    public function vectors(): array
    {
        return LcgStreamVectors::all();
    }
}

File: fixtures/scale/objc/large/Families/LcgStream/LcgStreamDescriptor.m
#import "LcgStreamDescriptor.h"
#import "LcgStreamCipher.h"
#import "LcgStreamVectors.h"

@implementation LcgStreamDescriptor

- (NSString *)family {
    return @"lcgstream";
}

- (NSString *)suite {
    return @"stream";
}

- (id<CLCipher>)cipher {
    return [[LcgStreamCipher alloc] init];
}

- (NSArray *)vectors {
    return [LcgStreamVectors all];
}

@end
